### DISCUSSION TITLE
Difficulty running the Caracal repository locally

### DISCUSSION THREAD

**Discussion by Moliki-Salman** [camicroscope/caMicroscope#758] [Q&A]
Hello, 
 @birm I am trying to setup the caracal repository locally and run it. I have cloned all the three required repository (caracal, Distro and caMicroscope) in a parent directory as instructed in the caracal readme.md. However, I am unable to run  bash ./setup_script  inside the caracal repo. whenever i try to run it, it throws " [ database ] : name not passed, set to default value (127.0.0.1)
[ database ] : name not passed, set to default value (camic)
mongo could not be found on path. Please ensure that mongo is installed and is on PATH". I have installed mongo but i am still unable to run the script. 
Please kindly guide me on what to do to solve this problem. 

Also, i want to clarify. The readme.md  of caracal  repo states that: "First clone the [Caracal (backup-dev branch)], but I could not see a branch named  backup-dev branch under Caracal. I only cloned Caracal. 

I look forward to your  guidance. Thank you

**Comment by birm** [comment]
Ah, you're using caracal without docker compose!  While we usually use docker compose since it's likely easier, you can do this manually.

Caracal expects mongo to be running somewhere. Configuration for caracal is done through environment variables. setup_script.sh is trying to help set those for you, so it takes some arguments as input. It'd be run `bash setup_script.sh (database host) (database name)`, where database host is whatever host you have mongo installed on, and database name is optional, "camic" should be fine.

If you don't have mongo set up somewhere else, you can [set it up on your machine directly](https://www.mongodb.com/docs/manual/installation/) or use a [container with ports mapped to localhost](https://www.howtogeek.com/devops/how-to-run-mongodb-in-a-docker-container/), and just run setup_script with default args only.

Backup dev is likely quite old, thanks for pointing that out!

I hope this helps; let us know if you have further questions or issues!!

**Reply by SanjanaSogimatt** [reply]
Hey @birm can I know on which PATH we are supposed to add mongo? As I have already added it on my path environment.

**Comment by SanjanaSogimatt** [comment]
Hey, @Moliki-Salman was the issue resolved for you? Because I'm still stuck on it

**Comment by Moliki-Salman** [comment]
@SanjanaSogimatt I  just figured it out. This was how I was able to resolve the issue: 
Although I had mongodb installed on my system, it was only running as a GUI, so I reinstalled it on  terminal for it  to enable  the necessary tools needed for the command line.

@birm I discovered that the mongo command line tool is no longer shipped with MongoDB version 6.0. This tool is deprecated and replaced by the new mongosh shell. Thus, by reinstalling mongodb, it will automatically install mongosh. This is why mongo was not recognised in PATH

 As regards, adding mongo to path: Inside the setup_script.sh, I changed all mongo to mongosh and all “mongo” to “mongosh”. Then I created camic database in  mongodb GUI. Then run bash setup_script.sh mongodb:localhost:27017/camic. Then run npm start 

I hope this might be helpful to you.

**Reply by SanjanaSogimatt** [reply]
Yup figured it out too @Moliki-Salman , but thanks

**Reply by birm** [reply]
This came up in another discussion for other reasons, but the shell script for caracal is _old_ as we haven't kept up with it much. It's neglected as I almost always end up deploying with docker.
Thanks for noticing this. I'll comment on the open issue (https://github.com/camicroscope/Caracal/issues/171) with your additional findings!

**Reply by gbengaoluwadahunsi** [reply]
nice and thanks to @Moliki-Salman  @birm  @SanjanaSogimatt
